README.md

```diff
@@ -1,2 +1,133 @@
+
+## Considerations for running SPARCRequest in Kubernetes
+
+### SPARCRequest Customization
+
+By using a second image we are able to modify our instance of SPARCRequest without needing to maintain a separate branch in source control.
+
+See the Image section below for a description how we build and manage our SPARCRequest instance.
+
+### Deployments
+
+We have two deployment workloads defined; one for the main application and a second for the delayed job that performs background tasks like sending emails.
+
+#### Main Deployment
+
+The main SPARCRequest application generally runs well inside Kubernetes, a single process Puma web server running the rails app. As seen in the [`deployment`](./k8s/deployment.yaml) manifest, we specify the resources, configuration, liveness, initContainers and any volumes required for persistence.
+
+The `initContainers` are used to perform database migrations and asset compilation.
+
+The volumes persist the things like attachments. We use a NFS  `PersistentVolume` to preserve these files beyond the lifespan of a pod.
+
+The `livenessProbe` informs the scheduler when the new pod is up and that it can terminate the old pod. This is important because the database migrations and asset compilation can take a bit of time and this reduces the time the application is unavailable.
+
+#### Delayed Job
+
+The challenging part of running SPARCRequest in a Kubernetes environment is running the Delayed Job as this is a separate process from SPARCRequest. To accommodate this process we run it as a [`Deployment`](./k8s/deployment_delayed_job.yaml) running the following command:
+
+```bash
+rails jobs:work
+```
+
+This can be seen in the [`deployment_delayed_job`](./k8s/deployment_delayed_job.yaml#L20) manifest. There is no reason it could not be a container in the same pod. We chose to run it as a separate pod to assist with troubleshooting and to isolate the main application in case the delayed job runs amok. This also means that we can restart it independently of the main application.
+
+
+### Configuration
+
+Much of SPARCRequest's configuration is either performed via a number of files in the `config` directory. To minimize the amount of modification of the configuration files we converted them to rely on environment variables. See the [`database.yml`](./custom/deps/sparc/database.yml) file as an example. By externalizing the configuration we can use the same code in any environment, only updating configuration as needed.
+
+Then by using a [`ConfigMap`](./k8s/config.yaml) and [`Secrets`](./k8s/secrets.yaml) we can deploy the same image to any of our environments (dev, stage, prod) and it will be configured appropriately. The [`Deployment`](./k8s/deployment.yaml) and other resources can then reference to same configuration. See the [`deployment.yaml`](./k8s/deployment.yaml#L38) for an example how the configuration is referenced.
```

Comment:
```suggestion
Then by using a [`ConfigMap`](./k8s/config.yaml) and [`Secrets`](./k8s/secrets.yaml) we can deploy the same image to any of our environments (dev, stage, prod) and it will be configured appropriately. The [`Deployment`](./k8s/deployment.yaml) and other resources can then reference the same configuration. See the [`deployment.yaml`](./k8s/deployment.yaml#L38) for an example of how the configuration is referenced.
```